// react-soundplayer/lib/components/index.js
'use strict';

var _interopRequireWildcard = function (obj) { return obj && obj.__esModule ? obj : { 'default': obj }; };

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _PlayButton2 = require('./PlayButton');

var _PlayButton3 = _interopRequireWildcard(_PlayButton2);

exports.PlayButton = _PlayButton3['default'];

var _NextButton2 = require('./NextButton');

var _NextButton3 = _interopRequireWildcard(_NextButton2);

exports.NextButton = _NextButton3['default'];

var _PrevButton2 = require('./PrevButton');

var _PrevButton3 = _interopRequireWildcard(_PrevButton2);

exports.PrevButton = _PrevButton3['default'];

var _Progress2 = require('./Progress');

var _Progress3 = _interopRequireWildcard(_Progress2);

exports.Progress = _Progress3['default'];

var _Timer2 = require('./Timer');

var _Timer3 = _interopRequireWildcard(_Timer2);

exports.Timer = _Timer3['default'];

var _Cover2 = require('./Cover');

var _Cover3 = _interopRequireWildcard(_Cover2);

exports.Cover = _Cover3['default'];

var _Icons2 = require('./Icons');

var _Icons3 = _interopRequireWildcard(_Icons2);

exports.Icons = _Icons3['default'];
